csharp/autobuilder/Semmle.Autobuild.Shared/BuildCommandAutoRule.cs

```diff
@@ -1,15 +1,23 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using Semmle.Util.Logging;
 
-namespace Semmle.Autobuild
+namespace Semmle.Autobuild.Shared
 {
     /// <summary>
     /// Auto-detection of build scripts.
     /// </summary>
-    class BuildCommandAutoRule : IBuildRule
+    public class BuildCommandAutoRule : IBuildRule
     {
+        private readonly Func<Autobuilder, Func<IDictionary<string, string>?, BuildScript>, BuildScript> withDotNet;
```

Comment:
`Func<Autobuilder, Func<IDictionary<string, string>?, BuildScript>, BuildScript>` could be turned into a delegate as it's written out 4 times.